### PR TITLE
Core - transaction payload that support new output and input lists.

### LINKED ISSUE #213: Core - transaction payload with new outputs

### PR DESCRIPTION
**Pull request by rebinmannady93**
# Description of change

Fixes #213

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

test_transaction.c

## Change checklist

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/core/models/payloads/transaction.c [hunk 1]
```diff
@@ -164,26 +134,44 @@ size_t tx_essence_serialize(transaction_essence_t* es, byte_t buf[]) {
...
-    free(es);
+  transaction_essence_t* es = tx_essence_new();
+
+  size_t offset = 0;
+
+  // transaction type
+  memcpy(&es->tx_type, &buf[offset], sizeof(uint8_t));
+  offset += sizeof(uint8_t);
+
+  es->inputs = utxo_inputs_deserialize(&buf[offset], buf_len - offset);
```
Comment (matjazv) [inline src/core/models/payloads/transaction.c:154]: Check for NULL `es->inputs` is missing.

File: src/core/models/payloads/transaction.c [hunk 2]
```diff
@@ -164,26 +134,44 @@ size_t tx_essence_serialize(transaction_essence_t* es, byte_t buf[]) {
...
+  offset += utxo_inputs_serialize_len(es->inputs);
+
+  es->outputs = utxo_outputs_deserialize(&buf[offset], buf_len - offset);
```
Comment (matjazv) [inline src/core/models/payloads/transaction.c:161]: Check for NULL `es->outputs` is missing.

File: src/core/models/payloads/transaction.c [hunk 3]
```diff
@@ -164,26 +134,44 @@ size_t tx_essence_serialize(transaction_essence_t* es, byte_t buf[]) {
...
+  offset += utxo_outputs_serialize_len(es->outputs);
+
+  // transaction type
+  memcpy(&es->payload_len, &buf[offset], sizeof(uint32_t));
+  offset += sizeof(uint32_t);
```
Comment (matjazv) [inline src/core/models/payloads/transaction.c]: Check for buffer length is missing.

File: src/core/models/payloads/transaction.c [hunk 4]
```diff
@@ -164,26 +134,44 @@ size_t tx_essence_serialize(transaction_essence_t* es, byte_t buf[]) {
...
+
+  if (es->payload_len > 0) {
+    // To Do - Deserialize indexation payload
   }
+
+return es;
 }
 
 void tx_essence_print(transaction_essence_t* es) {
```
Comment (matjazv) [inline src/core/models/payloads/transaction.c:184]: Should (indexation) payload also be printed?

**Review by matjazv — CHANGES_REQUESTED** [review]

File: src/core/models/payloads/transaction.c (hunk 1 above)
Comment (rebinmannady93) [reply src/core/models/payloads/transaction.c:154]: Done.

File: src/core/models/payloads/transaction.c (hunk 2 above)
Comment (rebinmannady93) [reply src/core/models/payloads/transaction.c:161]: Done.

File: src/core/models/payloads/transaction.c (hunk 3 above)
Comment (rebinmannady93) [reply src/core/models/payloads/transaction.c]: Done.

File: src/core/models/payloads/transaction.c (hunk 4 above)
Comment (rebinmannady93) [reply src/core/models/payloads/transaction.c:184]: Yes, curently method to print indexation is not implemented. We will add it while doing the changes for indexation payload. Added a TBD comment.

**Review by matjazv — APPROVED** [review]

**Review by oopsmonk — APPROVED** [review]

**Pull request merged**